### ISSUE TITLE
Canto hangs on session with manual annotations and many high-throughput annotations

### ISSUE THREAD

**Issue by manulera** [pombase/canto#2602]
Hi @kimrutherford I wanted to change some annotations from https://curation.pombase.org/pombe/curs/5724d1814ea9f342
and the page took several minutes to render the links in the phenotype annotations usable.

I think it's because it was loading annotations that seem to come some high-throughput input (they are not shown in the phenotype list). I wonder whether if when there are many htp annotations, we can either paginate them, or have a button to decide whether to load all / load more?

**Comment by ValWood** [comment]
That would be good, because we rarely, if ever need to see the htp annotations in Canto (since we can't edit them there).

**Comment by kimrutherford** [comment]
> I think it's because it was loading annotations that seem to come some high-throughput input

> That would be good, because we rarely, if ever need to see the htp annotations in Canto

So would a good initial step be to only show non-HTP annotations?  That's quite a quick fix.  Pagination or a show-all button will take longer.

**Comment by ValWood** [comment]
It seems an obvious thing to do that I now can't think why we loaded them into Canto? is there a reason that we haven't remembered?

**Comment by kimrutherford** [comment]
> It seems an obvious thing to do that I now can't think why we loaded them into Canto?

Just to be clear, the existing annotations aren't loaded into Canto.  They are queried from Chado and displayed as needed.

> is there a reason that we haven't remembered?

Probably part of the reason is that we didn't have much data initially so showing all existing annotations for a publication wasn't a problem.  Also initially we didn't make a distinction in Chado between LTP and HTP annotation, so it wouldn't have been possible to show just the LTP annotation.

**Comment by ValWood** [comment]
Of course. We needed to see any existing LTP annotations, so if we just don't load the HTP that would be perfect. We need to remember that we will not see the HTP localization annotations. I often filter these if they are shown to be incorrect based on detailed experiments (because most of the orfeome were overexpressed constructs). But  we should also be looking a the webpages to see that nothing is out of place so it should be OK.

**Comment by ValWood** [comment]
@manulera Manu as an aside I have not mentioned this yet:

If you spot any spurious GO annotations from the orfeome collection we remove them
pombe-embl/external_data/external-go-data
GO_ORFeome_localizations2.txt

but just to keep a record of what was removed from this dataset I copy them to here:
GO_ORFeome_localizations2_deleted.txt

probably unnecessary, but  because it it feels a bit uncomfortable to cherry pick a large dataset I felt that I should at least have a list of things removed.

**Comment by manulera** [comment]
I think some of the HTP is useful to be loaded, such as biogrid interactions, not to double-do them, so I would say not loading them at all should not be the way to go. For how often this problem would occur, it may not be worth doing if it takes significant time to add the feature.

**Comment by ValWood** [comment]
Good point, it would be useful to see the BIOGRD interactions.
Although, we had not really thought about how the BIOGRID interactions would integrate with our new GI workflow. We need to make the interactions again to connect them to phenotypes. But we don't want to submit redundant data to BIOGRID either.

**Comment by kimrutherford** [comment]
> and the page took several minutes to render the links in the phenotype annotations usable.

Fixed!

The reason for this page lock-ups wasn't what I imagined.  It turns out that the quick fix I had implemented for pombase/canto#1160 (6 years ago!) was causing the problem.  That fix was forcing the browser to do a lot more work than it should to in order to fill in the missing extension term names.

I've implemented a better solution now so that the browser doesn't lock up.  It still takes some seconds to load [5724d1814ea9f342](https://curation.pombase.org/pombe/curs/5724d1814ea9f342/ro) but it's then immediately usable.

This fix will improve the speed of all pages that have extensions containing terms, but it won't be noticeable for small sessions.

We still might want to think about ways of making big tables more user friendly, but that's a separate question.
